dvc/remote/ssh/connection.py

```diff
@@ -185,17 +185,18 @@ def download(self, src, dest, no_progress_bar=False, progress_title=None):
             self.sftp.get(src, dest, callback=pbar.update_to)
 
     def move(self, src, dst):
-        """Use `SFTP RENAME` operation when possible, otherwise,
-        fallback to copy and remove, since the former one doesn't
-        work across drives.
+        """Atomically move src to dst.
+
+        Moving is performed in two stages to make the whole operation atomic in
+        case src and dst are on different filesystems and actual physical
+        copying of data is happening.
         """
         self.makedirs(posixpath.dirname(dst))
 
-        try:
-            self.sftp.rename(src, dst)
-        except OSError:
-            self.copy(src, dst)
-            self.remove(src)
+        tmp = tmp_fname(dst)
+        self.copy(src, tmp)
```

Comment:
so, why do we do copy first if we can avoid it in certain cases?

Reply:
totally, @shcheklein , I was thinking that it was a better idea to always `copy` but there's a huge performance penalty.

changed it already to `sftp.rename` and fallback to `copy` only if it fails :slightly_smiling_face: